File: UI_Mobile_Performance/UI_Performance_android_java/app/src/main/java/com/example/uiperformance/MainActivity.java
package com.example.uiperformance;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.LinearSmoothScroller;
import androidx.recyclerview.widget.RecyclerView;

import android.graphics.PointF;
import android.os.Bundle;
import android.util.DisplayMetrics;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    private static final float MILLISECONDS_PER_INCH = 670f;

    RecyclerView recyclerView;
    Button startButton;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        final PerformanceAdapter adapter = new PerformanceAdapter(this);
        final LinearLayoutManager layoutManager  = new LinearLayoutManager(this);
        recyclerView = findViewById(R.id.recyclerview);
        startButton = findViewById(R.id.startButton);
        recyclerView.setLayoutManager(layoutManager);
        recyclerView.setHasFixedSize(true);
        recyclerView.setAdapter(adapter);
        startButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                final RecyclerView.SmoothScroller smoothScroller = new LinearSmoothScroller(MainActivity.this) {
                    @Override protected int getVerticalSnapPreference() {
                        return LinearSmoothScroller.SNAP_TO_START;
                    }
                    @Override
                    public PointF computeScrollVectorForPosition(int targetPosition) {
                        return super.computeScrollVectorForPosition(targetPosition);
                    }

                    @Override
                    protected float calculateSpeedPerPixel(DisplayMetrics displayMetrics) {
                        return MILLISECONDS_PER_INCH / displayMetrics.densityDpi;
                    }
                };
                smoothScroller.setTargetPosition(1000);
                layoutManager.startSmoothScroll(smoothScroller);
            }
        });
    }
}

File: Lottie_Performance/lottie_android/app/src/main/java/com/example/lottie_android/MainActivity.kt
package com.example.lottie_android

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.RecyclerView


class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val adapter = LottieRecyclerViewAdapter(this)
        val layoutManager = GridLayoutManager(this, 5)
        var recyclerView = findViewById<RecyclerView>(R.id.rvLottie)
        recyclerView.setLayoutManager(layoutManager)
        recyclerView.setHasFixedSize(true)
        recyclerView.setAdapter(adapter)
    }
}
